README.md

```diff
@@ -70,15 +119,279 @@ solidity: |-
+
+Connecting a synchronous blocking HTTP interface directly to an inherently asynchronous system like this can cause problems. The HTTP requester times out waiting for a response, but has no way to know if it should retry or if the transaction will eventually succeed. It cannot cancel the request.
+
+Providing a Messaging layer with at-least-once delivery and message ordering, allows the asynchronous nature of Ethereum to be reflected back to the remote application.
+
+Applications that have their own state stores are able to communicate over Messaging / Kafka with trivially simple JSON payloads to stream transactions into a scalable set of Ethereum nodes, and process the replies as they occur. The application can scale horizontally. Applications can also be decoupled from the Ethereum network with an integration technology like an Enterprise Service Bus (ESB).
+
+When spikes in workload occur that create a large queue of transactions that need to be fed into the Ethereum network at a lower rate, the kaleido-io/ethconnect bridge feeds them in at an optimal rate.
+
+### Ethereum Webhooks and the REST Receipt Store (MongoDB)
+
+Another key goal of having a robust Messaging layer under the covers is that the most trivial application can send messages into Ethereum reliably.
+
+- `POST` a [trivial YAML/JSON payload](#yaml-to-submit-a-transaction)
+  - to `/hook` to complete once the message is confirmed by Kafka (0.5s - tunable)
+  - to `/fasthook` to complete immediately when the message is sent to Kafka
+- Get an _immediate_ receipt
+   - No need to wait many seconds for a block to be cut
+   - No need to have complex retry handling
+
+This is ideal for Applications, Integration-as-a-Service (IaaS) platforms, and built-in SaaS integrations that support Webhooks for event submission.
+
+However, SaaS administrators and end-users still need to be able to track down what happened for a particular transaction, to diagnose problems, and correlate details of the Ethereum transaction with the business event.
+
+So kaleido-io/ethereum comes with a built-in receipt store, backed by MongoDB. It listens reliably for the replies over Kafka, and inserts each of them into the Database.
+
+It provides a trivially simple REST API:
+- `GET` `/reply/a789940d-710b-489f-477f-dc9aaa0aef77` to look for an individual reply
+- `GET` `/replies` to list the replies
+  - Ordered by time _received_ (not the order submitted) - listing the newest first
+  - `limit` and `skip` query parameters can be used to paginate the results
+
+A capped collection can be used in MongoDB to limit the storage. For example to store only the last 1000 replies received.
+
+### Scale and message ordering - nonce management
```

Comment:
Nonce Management for scale and message ordering.

Nonce Management is the more interesting bit of the headline for me.